renku/core/dataset/dataset_add.py

```diff
@@ -155,6 +159,21 @@ def _download_files(
     **kwargs,
 ) -> List["DatasetAddMetadata"]:
     """Process file URLs for adding to a dataset."""
+    if dataset.storage and dataset.storage.lower().startswith("s3://"):
+        if len(urls) == 0:
+            raise errors.ParameterError("No URL is specified")
+        if len(urls) > 1:
+            raise errors.ParameterError("Can only add 1 uri with S3 datasets.")
+        if not urls[0].lower().startswith("s3://"):
+            raise errors.ParameterError("Only files from S3 buckets can be added to S3 datasets.")
+        if sources:
+            raise errors.ParameterError("Can not use '--source' with S3 datasets.")
+        if len(dataset.files) > 0:
```

Comment:
This is too limiting: It means that we can only add one file to a dataset. Is there a reason not to allow adding more files?

Reply:
This is a limitation if you are mounting. If you are copying it becomes easier to handle. I will remove it and we can deal with it in the PR about copying. 

IMO copying kind of defeats the whole purpose of using S3 to some extent. Because the biggest motivation for using s3 is to not need the same storage in the session (or your local machine).

Reply:
Copying is for use-cases where a user want to have faster access to data; mounting is more space-efficient but we always need to stream data from the external storage (unless the underlying library implements a caching mechanism).

Reply:
Ok yeah I removed this.